src/main/scala/temple/generate/target/openapi/OpenAPIBuilder.scala

```diff
@@ -0,0 +1,149 @@
+      val maxLength = max.map("maxLength" -> _.asJson)
+      OpenAPISimpleType("string", Seq(minLength, maxLength).flatten: _*)
+    case IntType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "int64" else "int32", Seq(minimum, maximum).flatten: _*)
+    case FloatType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "double" else "float", Seq(minimum, maximum).flatten: _*)
+    case ForeignKey(references) =>
+      OpenAPISimpleType("number", "int32", "description" -> s"Reference to $references ID".asJson)
+  }
+
+  private def generateItemType(attributes: Map[String, Attribute]): OpenAPIObject = OpenAPIObject(
+    attributes.iterator
+      .filter { case _ -> attribute => !isServerAttribute(attribute) }
+      .map { case str -> attribute => str -> attributeToOpenAPIType(attribute) }
+      .to(attributes.mapFactory),
+  )
+
+  private def generateItemInputType(attributes: Map[String, Attribute]): OpenAPIObject = OpenAPIObject(
+    attributes.iterator
+      .filter { case _ -> attribute => isClientAttribute(attribute) }
+      .map { case str -> attribute => str -> attributeToOpenAPIType(attribute) }
+      .to(attributes.mapFactory),
+  )
+
+  private def jsonContent(mediaTypeObject: MediaTypeObject) = Map("application/json" -> mediaTypeObject)
+
+  def addPaths(service: Service): this.type = {
+    val lowerName       = service.name.toLowerCase
+    val capitalizedName = service.name.capitalize
+    val tags            = Seq(capitalizedName)
+    service.endpoints.foreach {
+      case ReadAll =>
+        path(s"/$lowerName/all") += HTTPVerb.Get -> Path(
+            s"Get a list of every $lowerName",
+            tags = tags,
+            responses = Map(
+              200 -> Literal(
+                jsonContent(MediaTypeObject(OpenAPIArray(generateItemType(service.attributes)))),
+                s"$capitalizedName list successfully fetched",
+              ),
+              500 -> Response.Ref(useError(500)),
+            ),
+          )
+      case Create =>
+        path(s"/$lowerName") += HTTPVerb.Post -> Path(
+            s"Register a new $lowerName",
+            tags = tags,
+            requestBody = Some(Literal(jsonContent(MediaTypeObject(generateItemInputType(service.attributes))))),
+            responses = Map(
+              200 -> Literal(
+                jsonContent(MediaTypeObject(generateItemType(service.attributes))),
+                s"$capitalizedName successfully created",
+              ),
+              400 -> Response.Ref(useError(400)),
+              500 -> Response.Ref(useError(500)),
+            ),
+          )
+      case Read   => // TODO in future PR
+      case Update => // TODO in future PR
+      case Delete => // TODO in future PR
```

Comment:
Would each of these be better served in their own functions? Does mean I can’t prepare `lowerName`, `capitalizedName` etc in a similar place for all of them

Reply:
Maybe, then they just take it as a parameter? Happy either way though